### PR TITLE
Add tests for inclusive boundary rules

### PR DESCRIPTION
**Pull request by goodwillpunning**
This commit updates the validation logic for the `Bounds` class. It also adds a valuable test case for inclusive boundary rules.

### CODE REVIEW

File: src/main/scala/com/databricks/labs/validation/utils/Structures.scala [hunk 1]
```diff
@@ -27,8 +27,8 @@ object Structures {
                      lowerInclusive: Boolean = false,
                      upperInclusive: Boolean = false) {
     def validationLogic(c: Column): Column = {
-      val lowerLogic = if (lowerInclusive) c <= lower else c < lower
-      val upperLogic = if (upperInclusive) c >= upper else c > upper
+      val lowerLogic = if (lowerInclusive) c >= lower else c > lower
```
Comment (GeekSheikh) [inline src/main/scala/com/databricks/labs/validation/utils/Structures.scala:30]: NICE CATCH!! Tests are awesome!! HHAAHA

**Review by GeekSheikh — APPROVED** [review]
lgtm
